### ISSUE TITLE
ImmediateGeometry Node does not color the points 

### ISSUE THREAD

**Issue by dodgyville** [godotengine/godot#23045]
**Godot version:**
3.1 master from the last few days

**OS/device including version:**
Intel HD 5000 (macbook air)

**Issue description:**
I'm using ImmediateGeometry to draw a list of points. I believe I can set the colour of the point similar to an opengl call by using IM.set_color inside the IM.begin call.

However, the points render black in the game. I think it could be one of several problems:
1. my misunderstanding the use of set_color
2. I'm using gles2 as it's an older laptop and perhaps it's not supported
3. There's a bug where ambient light does not affect the points
4. There's a bug where no lights affect the points (I added an omnilight to no effect)

I think this is related to #10024 which describes a similar problem on godot 2.

**Steps to reproduce:**
```
var point_size = 5
var im = ImmediateGeometry.new()
add_child(im)
var m = SpatialMaterial.new()
m.flags_use_point_size = true
m.params_point_size = point_size
im.set_material_override(m)
im.clear()
im.begin(Mesh.PRIMITIVE_POINTS, null)
im.set_color(Color(0,1,1))
for p in pts: #list of Vector3s
    im.add_vertex(p)
im.end()
```

**Comment by Zylann** [comment]
I think you forgot to use `vertex color as albedo`, this works for me in 3.1 alpha1:
```gdscript
	var point_size = 5
	var im = ImmediateGeometry.new()
	add_child(im)
	var m = SpatialMaterial.new()
	m.flags_use_point_size = true
	m.params_point_size = point_size
	m.vertex_color_use_as_albedo = true # <-- THIS
	im.set_material_override(m)
	im.clear()
	im.begin(Mesh.PRIMITIVE_POINTS, null)
	im.set_color(Color(0,1,1))
	for i in 100:
	    im.add_vertex(Vector3(rand_range(-1, 1), rand_range(-1, 1), rand_range(-1, 1)))
	im.end()
```

**Comment by dodgyville** [comment]
Thanks, it works on linux but not my old mac laptop using gles2 but I can live with that!

**Comment by honghaoli** [comment]
same here for macOS, but the color is not black, always gray and light green-ish.

**Comment by afonsolage** [comment]
Just to note it worked fine on Windows.

**Comment by andrew-wilkes** [comment]
In the material, I need flags_unshaded = true in order to see Albedo otherwise get black.

**Comment by KoBeWi** [comment]
Can anyone still reproduce this bug in Godot 3.2.3 or any later release?

**Comment by WilliamTambellini** [comment]
@KoBeWi  yes I can repro with 3.2.3 under Linux: set_color() does nt seem to do anything, points are just always black.

**Comment by Calinou** [comment]
@WilliamTambellini Did you enable "unshaded" on the SpatialMaterial used for the ImmediateGeometry node?

**Comment by Zylann** [comment]
Don't forget `vertex_color_use_as_albedo` otherwise it will not have any effect

**Comment by Calinou** [comment]
@dodgyville Can you (or anyone else) still reproduce this bug in [Godot 3.4.2](https://godotengine.org/article/maintenance-release-godot-3-4-2) or any later release?

If yes, please ensure that an up-to-date Minimal Reproduction Project (MRP) is included in this report (a MRP is a zipped Godot project with the minimal elements necessary to reliably trigger the bug). You can upload ZIP files in an issue comment with a drag and drop.
